docs/source/autoscaling.rst

```diff
@@ -2,55 +2,40 @@
 Autoscaling PaaSTA Instances
 ====================================
 
-The section below introduces autoscaling in Marathon.
-For services in Kubernetes, the following configurations are allowed in configuration files,
-for compatibility with marathon.yaml; but not all of them are respected.
-The mechanism is also different. If you are creating a new service on Kubernetes or are modifying
-autoscaling configurations for your services that already running on Kubernetes, please
-check :ref:`here <hpa>`.
-
-PaaSTA allows programmatic control of the number of marathon tasks a service has.
-It does this by using Zookeeper as a data store to record the number of tasks it
-thinks should be running for each instance of a service on a per-cluster basis.
-
+PaaSTA allows programmatic control of the number of replicas (pods) a service has.
+It uses Kubernetes' Horizontal Pod Autoscaler (HPA) to watch a service's load and scale up or down.
 
 How to use autoscaling
 ======================
 
 Enabling autoscaling
 --------------------
 
-In order to use autoscaling, edit your ``marathon-*.yaml`` files in your soa
+In order to use autoscaling, edit your ``kubernetes-*.yaml`` files in your soa
 configs and add a ``min_instances`` and a ``max_instances`` attribute and
-remove the ``instances`` attribute from each instance you want to autoscale.
+(optionally) remove the ``instances`` attribute from each instance you want to autoscale.
```

Comment:
is this actually optional these days? I thought that `instances` took precedence and would prevent autoscaling

Reply:
ah, I see - we ignore `instances` if `max_instances` is set! this seems kinda confusing from a user perspective, maybe we should make the soaconfigs validator prohibit having instances + max_instances at some point so that its always obvious what will happen?

Reply:
Yeah, it's weird. It's been like this since we first introduced min_instances/max_instances. We needed to resolve the ambiguity one way or the other because the way we use yaml anchors doesn't make it easy to delete keys.

I've proposed fixing this by making `instances` into a Union of int and a TypedDict with `min` and `max` keys.